File: frontend/src/api/livestockApi.js
import apiClient from './axios'

export function getLivestockByFarm(farmId) {
  return apiClient.get(`/api/farms/${farmId}/livestock`).then((res) => res.data)
}

export function getLivestockById(id) {
  return apiClient.get(`/api/livestock/${id}`).then((res) => res.data)
}

export function getLivestockHealth(id) {
  return apiClient.get(`/api/livestock/${id}/health`).then((res) => res.data)
}

export function createLivestock({ tagNumber, species, dateOfBirth, farmId }) {
  return apiClient
    .post('/api/livestock', { tagNumber, species, dateOfBirth, farmId })
    .then((res) => res.data)
}

export function updateLivestock(id, { species, dateOfBirth, status }) {
  return apiClient
    .patch(`/api/livestock/${id}`, { species, dateOfBirth, status })
    .then((res) => res.data)
}

File: mobile/lib/api/livestock_api.dart
import '../core/api_client.dart';
import '../core/json.dart';
import '../models/livestock.dart';
import '../models/livestock_health.dart';

class LivestockApi {
  static Future<Livestock> getById(int id) async {
    final data = await ApiClient.instance.get<dynamic>('/api/livestock/$id');
    return Livestock.fromJson(asMap(data));
  }

  static Future<LivestockHealth> getHealth(int id) async {
    final data = await ApiClient.instance.get<dynamic>('/api/livestock/$id/health');
    return LivestockHealth.fromJson(asMap(data));
  }

  static Future<Livestock> create({
    required String tagNumber,
    required String species,
    String? dateOfBirth,
    required int farmId,
  }) async {
    final data = await ApiClient.instance.post<dynamic>(
      '/api/livestock',
      data: {
        'tagNumber': tagNumber,
        'species': species,
        'dateOfBirth': (dateOfBirth == null || dateOfBirth.isEmpty) ? null : dateOfBirth,
        'farmId': farmId,
      },
    );
    return Livestock.fromJson(asMap(data));
  }

  static Future<Livestock> update(
    int id, {
    required String species,
    required String dateOfBirth,
    required String status,
  }) async {
    final data = await ApiClient.instance.patch<dynamic>(
      '/api/livestock/$id',
      data: {'species': species, 'dateOfBirth': dateOfBirth, 'status': status},
    );
    return Livestock.fromJson(asMap(data));
  }
}
